fdbcli/ExpensiveDataCheckCommand.actor.cpp

```diff
@@ -34,48 +34,46 @@ namespace fdb_cli {
 // The command is used to send a data check request to the specified process
 // The check request is accomplished by rebooting the process
 
-ACTOR Future<bool> expensiveDataCheckCommandActor(
-    Reference<IDatabase> db,
-    Reference<ITransaction> tr,
-    std::vector<StringRef> tokens,
-    std::map<Key, std::pair<Value, ClientLeaderRegInterface>>* address_interface) {
+ACTOR Future<bool> expensiveDataCheckCommandActor(Reference<IDatabase> db,
+                                                  Reference<ITransaction> tr,
+                                                  std::vector<StringRef> tokens,
+                                                  std::set<std::string>* addresses) {
 	state bool result = true;
 	if (tokens.size() == 1) {
-		// initialize worker interfaces
-		wait(getWorkerInterfaces(tr, address_interface));
+		wait(fetchAndUpdateWorkerInterfaces(db, addresses));
 	}
 	if (tokens.size() == 1 || tokencmp(tokens[1], "list")) {
-		if (address_interface->size() == 0) {
+		if (addresses->size() == 0) {
 			printf("\nNo addresses can be checked.\n");
-		} else if (address_interface->size() == 1) {
+		} else if (addresses->size() == 1) {
 			printf("\nThe following address can be checked:\n");
 		} else {
-			printf("\nThe following %zu addresses can be checked:\n", address_interface->size());
+			printf("\nThe following %zu addresses can be checked:\n", addresses->size());
 		}
-		for (auto it : *address_interface) {
-			printf("%s\n", printable(it.first).c_str());
+		for (const auto& addr : *addresses) {
+			printf("%s\n", printable(addr).c_str());
 		}
 		printf("\n");
 	} else if (tokencmp(tokens[1], "all")) {
-		state std::map<Key, std::pair<Value, ClientLeaderRegInterface>>::const_iterator it;
-		for (it = address_interface->cbegin(); it != address_interface->cend(); it++) {
-			int64_t checkRequestSent = wait(safeThreadFutureToFuture(db->rebootWorker(it->first, true, 0)));
+		state std::set<std::string>::iterator it;
+		for (it = addresses->begin(); it != addresses->end(); ++it) {
+			int64_t checkRequestSent = wait(safeThreadFutureToFuture(db->rebootWorker(*it, true, 0)));
 			if (!checkRequestSent) {
 				result = false;
-				fprintf(stderr, "ERROR: failed to send request to check process `%s'.\n", it->first.toString().c_str());
+				fprintf(stderr, "ERROR: failed to send request to check process `%s'.\n", it->c_str());
 			}
 		}
-		if (address_interface->size() == 0) {
+		if (addresses->size() == 0) {
 			fprintf(stderr,
 			        "ERROR: no processes to check. You must run the `expensive_data_check’ "
 			        "command before running `expensive_data_check all’.\n");
 		} else {
-			printf("Attempted to kill and check %zu processes\n", address_interface->size());
+			printf("Attempted to kill and check %zu processes\n", addresses->size());
 		}
 	} else {
 		state int i;
 		for (i = 1; i < tokens.size(); i++) {
```

Comment:
Use `std::map::count` (or `std::map::find` if you're not into that whole brevity thing)

Reply:
Forgot this, fixed this